### PR TITLE
Add a refactoring to expand all editor placeholders that can be converted to trailing closures

### PR DESCRIPTION
**Pull request by ahoppen**
Also allow expansion of trailing placeholders that only have a type and no text.

Companion of https://github.com/apple/sourcekit-lsp/pull/1072

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Tests/SwiftRefactorTest/ExpandEditorPlaceholderTests.swift [hunk 1]
```diff
@@ -12,13 +12,17 @@
...
+@_spi(Testing) import SwiftRefactor
 import SwiftSyntax
 import SwiftSyntaxBuilder
 import XCTest
 import _SwiftSyntaxTestSupport
 
-final class ExpandEditorPlaceholderTest: XCTestCase {
+fileprivate let closurePlaceholder = wrapInPlaceholder("T##closure##() -> Void")
+fileprivate let voidPlaceholder = wrapInPlaceholder("T##code##Void")
+fileprivate let intPlaceholder = wrapInPlaceholder("T##Int##Int")
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply Tests/SwiftRefactorTest/ExpandEditorPlaceholderTests.swift:23]: Ah, let’s just keep them.

**Comment by ahoppen** [comment]
There was decent coverage for it because `ExpandEditorPlaceholder` calls into `ExpandEditorPlaceholdersToTrailingClosures` but I added some dedicated tests as well.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SwiftRefactor/ExpandEditorPlaceholder.swift [hunk 2]
```diff
@@ -132,28 +120,68 @@ public struct ExpandEditorPlaceholder: EditRefactoringProvider {
...
+/// ## Before
+/// ```swift
+/// foo(
+///   arg: <#T##Int#>,
+///   firstClosure: <#T##(Int) -> String##(Int) -> String##(_ someInt: Int) -> String#>,
+///   secondClosure: <#T##(Int) -> String##(Int) -> String##(_ someInt: Int) -> String#>
+/// )
+/// ```
+///
+/// ## `closure3` or `closure4` Expansion
```
Comment (bnbarham) [inline Sources/SwiftRefactor/ExpandEditorPlaceholder.swift:151]: "`foo` expansion right?

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**